virl2_client/models/lab.py

```diff
@@ -955,16 +955,17 @@ def _create_link_local(
 
     @check_stale
     @locked
-    def connect_two_nodes(self, node1: Node, node2: Node) -> Link:
+    def connect_two_nodes(self, node1: Node, node2: Node, index: int = 0) -> Link:
         """
         Connect two nodes within a lab.
 
         :param node1: The first node object.
         :param node2: The second node object.
+        :param index: An optional starting interface index (default: 0).
         :returns: The created link.
         """
-        iface1 = node1.next_available_interface() or node1.create_interface()
-        iface2 = node2.next_available_interface() or node2.create_interface()
+        iface1 = node1.next_available_interface(index) or node1.create_interface()
+        iface2 = node2.next_available_interface(index) or node2.create_interface()
```

Comment:
if the caller requests that interface at least `index` is used, then create_interface should also be passed that index.
Also, if you e.g. want to skip the unused xrv9k interfaces, then the next_available_interface function might want to do it based on the interface label. Then, if you want to connect interface 3 and one of the nodes is an external connector, then your current code will work only if the node does not yet have any interface and create_interface is called.